controller/src/main/java/io/pravega/controller/eventProcessor/impl/EventProcessorCell.java

```diff
@@ -100,7 +103,16 @@ protected final void run() throws Exception {
 
             while (isRunning()) {
                 try {
-                    event = reader.readNextEvent(defaultTimeout);
+                    try {
+                        event = reader.readNextEvent(defaultTimeout);
+                    } catch (HostStoreException e) {
+                        // In the case of a SessionExpiredException, we don't want to restart (the Controller itself will).
+                        if (e.getCause() instanceof KeeperException.SessionExpiredException) {
```

Comment:
As part of memory footprint improvements done in controller, we changed the mechanics of ZKHostStore to not make zk call for each look up. Instead we use NodeCache which populates a local map. 

So SessionExpiredException will not be thrown anymore. 

Secondly, in case of an In-Memory deployment, there is no notion of a session. So any SessionSpecific check here breaks the abstraction.

Reply:
Thanks @shiveshr, I have removed this unnecessary exception handling.